Hippo.Web/ClientApp/src/components/ClusterAdmin/SoftwareRequestForm.tsx

```diff
@@ -0,0 +1,205 @@
+
+const defaultSoftwareRequestModel = {
+  clusterName: "",
+  email: "",
+  accountName: "",
+  softwareTitle: "",
+  softwareLicense: "",
+  softwareHomePage: "",
+  benefitDescription: "",
+  additionalInformation: "",
+} as unknown as SoftwareRequestModel;
+
+export const SoftwareRequestForm = () => {
+  const [context] = useContext(AppContext);
+  const [notification, setNotification] = usePromiseNotification();
+  const [softwareRequestModel, setSoftwareRequestModel] =
+    useState<SoftwareRequestModel>({
+      ...defaultSoftwareRequestModel,
+      email: context.user.detail.email,
+      accountName: context.user.detail.kerberos,
+    });
+
+  const handleSubmit = async () => {
+    const req = authenticatedFetch(`/api/software/requestinstall`, {
+      method: "POST",
+      body: JSON.stringify(softwareRequestModel),
+    });
+
+    setNotification(req, "Sending", "Request sent.", async (r) => {
+      if (r.status === 400) {
+        const errors = await parseBadRequest(response);
+        return errors;
+      } else {
+        return "An error happened, please try again.";
+      }
+    });
+
+    const response = await req;
+
+    if (response.ok) {
+      setSoftwareRequestModel({
+        ...defaultSoftwareRequestModel,
+        email: context.user.detail.email,
+        accountName: context.user.detail.kerberos,
+      });
+    }
+  };
+
+  return (
+    <div className="row justify-content-center">
+      <div className="col-md-8">
+        <h3>
+          Welcome,{" "}
+          <span className="status-color">{context.user.detail.firstName}</span>
+        </h3>
+        <p>
+          Please enter the following details to request a custom installation of
+          software...
+        </p>
+        <hr />
+        <div className="form-group">
+          <label>Cluster Name</label>
+          <select
+            className="form-control"
+            value={softwareRequestModel.clusterName}
+            onChange={(e) =>
+              setSoftwareRequestModel({
+                ...softwareRequestModel,
+                clusterName: e.target.value as ClusterName,
+              })
+            }
+          >
+            <option value="">Select a cluster...</option>
+            {ClusterNames.map((cluster) => (
+              <option key={cluster} value={cluster}>
+                {cluster}
+              </option>
+            ))}
+          </select>
+          <p className="form-helper">Select the name of the cluster.</p>
+        </div>
+        <div className="form-group">
+          <label>Email</label>
+          <input
+            type="email"
+            className="form-control"
+            value={softwareRequestModel.email}
+            onChange={(e) =>
+              setSoftwareRequestModel({
+                ...softwareRequestModel,
+                email: e.target.value,
+              })
+            }
+          />
+          <p className="form-helper">Enter your email address.</p>
```

Comment:
Should we allow them to do this, or get this info from IAM?

Reply:
It defaults to the value from IAM